Source/Core/VideoCommon/AbstractTexture.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+#include "Common/MathUtil.h"
+#include "VideoCommon/TextureConfig.h"
+#include "VideoCommon/VideoCommon.h"
+
+class AbstractTexture
+{
+public:
+  explicit AbstractTexture(const TextureConfig& c);
+  virtual ~AbstractTexture();
+  virtual void Bind(unsigned int stage) = 0;
+  virtual bool Save(const std::string& filename, unsigned int level);
+
+  virtual void CopyRectangleFromTexture(const AbstractTexture* source,
+                                        const MathUtil::Rectangle<int>& srcrect,
+                                        const MathUtil::Rectangle<int>& dstrect) = 0;
+  virtual void Load(u32 level, u32 width, u32 height, u32 row_length, const u8* buffer,
+                    size_t buffer_size) = 0;
+  virtual void FromRenderTarget(bool is_depth_copy, const EFBRectangle& src_rect,
```

Comment:
I'd be fine with "CopyRegionToTexture" or "CopyRegionToCacheEntry" because this is eventually going to include XFBs as well.  I'm fine with moving this for now, as being on AbstractTexture didn't really help or hurt us.